### ISSUE TITLE
`unused_parens` false-positive on mutable binding inside &-pattern

### ISSUE THREAD

**Issue by sinkuu** [rust-lang/rust#64064]
```rust
for &(mut _x) in &[1, 2, 3] {
    // _x: mutable binding to a copied element
}
```

`unused_parens` suggests removing parens, but doing so will turn the pattern into a `&mut` pattern.

```
warning: unnecessary parentheses around pattern
 --> test.rs:2:10
  |
2 |     for &(mut _x) in &[1, 2, 3] {
  |          ^^^^^^^^ help: remove these parentheses
  |
```

**Comment by Centril** [comment]
Duplicated of https://github.com/rust-lang/rust/issues/55342.
